ts/Core/Series/Series.ts

```diff
@@ -816,6 +816,19 @@ class Series {
             ),
             userPlotOptionsType = userPlotOptions[this.type] || {};
 
+        // When plot options have multiple data labels, the default data label
+        // plot options are not merged (#21928).
+        const dataLabelOptions = typeOptions.dataLabels;
+        if (dataLabelOptions instanceof Array) {
```

Comment:
I'm thinking about using `splat` function to simplify logic a little bit. Did you consider using `splat` and operating always on the array?

Reply:
Right, got it!